src/main/java/de/tum/in/www1/artemis/service/ModelAssessmentKnowledgeService.java

```diff
@@ -0,0 +1,48 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.Set;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.ModelAssessmentKnowledge;
+import de.tum.in.www1.artemis.domain.modeling.ModelingExercise;
+import de.tum.in.www1.artemis.repository.ModelAssesmentKnowledgeRepository;
+import de.tum.in.www1.artemis.repository.ModelingExerciseRepository;
+
+@Service
+public class ModelAssessmentKnowledgeService {
+
+    private final ModelAssesmentKnowledgeRepository modelAssesmentKnowledgeRepository;
+
+    private final ModelingExerciseRepository modelingExerciseRepository;
+
+    public ModelAssessmentKnowledgeService(ModelAssesmentKnowledgeRepository modelAssesmentKnowledgeRepository, ModelingExerciseRepository modelingExerciseRepository) {
+        this.modelAssesmentKnowledgeRepository = modelAssesmentKnowledgeRepository;
+        this.modelingExerciseRepository = modelingExerciseRepository;
+    }
+
+    /**
+     * delete only when no exercises use the knowledge
+     *
+     * @param knowledgeId
+     */
+    public void deleteKnowledge(Long knowledgeId) {
+        Set<ModelingExercise> exerciseSet = modelingExerciseRepository.findAllByKnowledgeId(knowledgeId);
+        // If no other exercises use the same knowledge then remove knowledge
+        if (exerciseSet.isEmpty()) {
+            modelAssesmentKnowledgeRepository.deleteById(knowledgeId);
+        }
```

Comment:
Would we want to throw an exception or log in cases when the `exerciseSet` is not empty?

Reply:
Hmm, I don't think an exception would be needed since in case the `exerciseSet` is not empty, this means we are still using the knowledge and therefore we just don't delete it 🙂

Reply:
I see. Then naming is a little bit confusing as I would expect the delete function to succeed or fail with a certain error otherwise. Delete function silently doing nothing is a little bit misleading.